c2_components/src/mfx_c2_encoder_component.cpp

```diff
@@ -979,7 +983,9 @@ mfxStatus MfxC2EncoderComponent::InitVPP(C2FrameData& buf_pack)
 
         mfx_res = m_vpp.Init(&param);
         m_inputVppType = param.conversion;
-    } else {
+    }
+        break;
+    default: {
         uint32_t width, height, format, stride, igbp_slot, generation;
         uint64_t usage, igbp_id;
         android::_UnwrapNativeCodec2GrallocMetadata(c_graph_block->handle(), &width, &height, &format, &usage,
```

Comment:
Is it possible we move this logic to Line 966. If the derived format is tiled Y , the default memory allocation is using video memory. If the derived format is linear, we use system memory allocation here.

Thanks,
Tianmi